rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/BaseWorkspaceService.java

```diff
@@ -72,13 +84,24 @@ public void initialize(ClientCapabilities clientCap, @Nullable List<WorkspaceFol
 
         var clientWorkspaceCap = clientCap.getWorkspace();
 
-        if (clientWorkspaceCap != null && Boolean.TRUE.equals(clientWorkspaceCap.getWorkspaceFolders())) {
-            var workspaceCap = new WorkspaceFoldersOptions();
-            workspaceCap.setSupported(true);
-            workspaceCap.setChangeNotifications(true);
-            capabilities.setWorkspace(new WorkspaceServerCapabilities(workspaceCap));
+        WorkspaceServerCapabilities workspaceCapabilities = new WorkspaceServerCapabilities();
+        if (clientWorkspaceCap != null) {
+            if (clientWorkspaceCap.getWorkspaceFolders()) {
+                var folderOptions = new WorkspaceFoldersOptions();
+                folderOptions.setSupported(true);
+                folderOptions.setChangeNotifications(true);
+                workspaceCapabilities.setWorkspaceFolders(folderOptions);
+            }
+
+            if (clientWorkspaceCap.getFileOperations().getDidRename()) {
+                var fileOperationCapabilities = new FileOperationsServerCapabilities();
+                fileOperationCapabilities.setDidRename(new FileOperationOptions(
+                    List.of(new FileOperationFilter(new FileOperationPattern("**")))));
```

Comment:
This is an example of the difference between rascal and parametric. We don't want rascal to receive events for all renames, only for the `**/*.rsc` changes.